akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -52,8 +58,17 @@ object CircuitBreaker {
    * @param callTimeout [[scala.concurrent.duration.FiniteDuration]] of time after which to consider a call a failure
    * @param resetTimeout [[scala.concurrent.duration.FiniteDuration]] of time after which to attempt to close the circuit
    */
-  def create(scheduler: Scheduler, maxFailures: Int, callTimeout: FiniteDuration, resetTimeout: FiniteDuration): CircuitBreaker =
+  def create(
+    scheduler:    Scheduler,
+    maxFailures:  Int,
+    callTimeout:  FiniteDuration,
+    resetTimeout: FiniteDuration): CircuitBreaker =
     apply(scheduler, maxFailures, callTimeout, resetTimeout)
+
+  def exceptionAsFailure[T]: Try[T] ⇒ Boolean = {
```

Comment:
`BiFunction` is like `scala.Function2`, i.e. 2 parameters. Question is how we should represent `Try`? As two `Optional`? `BiFunction[Optional[T], Optional[Throwable]]`?

Perhaps we can use `akka.actor.Status` to represent the `Try` in the Java API. It has `Sucess` and `Failure` that are consumable from Java.